akd/tools/search/semantic_scholar_search.py

```diff
@@ -171,6 +194,9 @@ async def _fetch_search_page(
                 logger.debug("Using API Key.")
 
         try:
+            # Apply rate limiting before making the request
+            await self.rate_limiter.acquire()
```

Comment:
See the usage section to understand this. Basically, we first acquire the lock to make ready for the requests (both async and sync).